flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseConnectorITCase.java

```diff
@@ -469,11 +470,13 @@ private void verifyHBaseLookupJoin(boolean async) {
                         + TEST_TABLE_1
                         + " FOR SYSTEM_TIME AS OF src.proc as h ON src.a = h.rowkey";
         Iterator<Row> collected = tEnv.executeSql(dimJoinQuery).collect();
-        List<String> result =
-                Lists.newArrayList(collected).stream()
-                        .map(Row::toString)
-                        .sorted()
-                        .collect(Collectors.toList());
+        List<String> result = new ArrayList<>();
+        for (Iterator<Row> it = collected; it.hasNext(); ) {
+            Row row = it.next();
+            result.add(row.toString());
+        }
+        Collections.sort(result);
+        result = new ArrayList<>(result);
```

Comment:
I think there is no need for a new arrayList, we can just reuse `result`

And similar thoughts about `StreamSupport`

```suggestion
        List<String> result =
                StreamSupport.stream(
                        Spliterators.spliteratorUnknownSize(collected, Spliterator.ORDERED),
                                false)
                        .map(Row::toString)
                        .sorted()
                        .collect(Collectors.toList());
```

Reply:
Same as my other comment.